### PR TITLE
New: Program::get_used_qubits()

### PR DESCRIPTION
**Pull request by notmgsk**

### CODE REVIEW

File: src/program/mod.rs [hunk 1]
```diff
@@ -367,4 +387,29 @@ DECLARE ec BIT
...
+    1,2
+I 0
+";
+        let program = Program::from_str(input).unwrap();
+        let expected = vec![Qubit::Fixed(0), Qubit::Variable("q".to_string())]
+            .into_iter()
+            .collect::<HashSet<_>>();
+        let actual = program.get_used_qubits();
+        assert_eq!(expected, actual);
+    }
```
Comment (nilslice) [inline src/program/mod.rs:414]: I may be missing it, but consider also include a qubit operation in the test Quil program that would _not_ be returned from the `get_used_qubits` method to check that this behavior is correct and continues to be later.

**Review by nilslice — CHANGES_REQUESTED** [review]

File: src/program/mod.rs (hunk 1 above)
Comment (nilslice) [reply src/program/mod.rs:414]: Thinking back on this, I'm not sure my request makes sense! Happy to connect on this offline 😆

**Comment by kalzoo** [comment]
I fumbled the merge-and-delete of `feat/improve-dot-graphs` - @notmgsk if you'd like to see this merged please just open a new MR against `main`.

File: src/program/mod.rs (hunk 1 above)
Comment (notmgsk) [reply src/program/mod.rs:414]: Yeah, not sure I can think of a case where that would happen.

**Pull request closed without merging**